packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts

```diff
@@ -75,43 +75,38 @@ export function useUploadFiles({
       };
 
       if (file) {
-        const {
-          file: processedFile,
-          key: processedKey,
-          ...processedRest
-        } = isFunction(processFile)
-          ? processFile({ file, key })
-          : { file, key };
-
-        onUploadStart?.({ key });
-
-        if (isResumable) {
-          const uploadTask = uploadFile({
-            ...processedRest,
-            file: processedFile,
-            fileName: path + processedKey,
-            isResumable: true,
-            level: accessLevel,
-            completeCallback: onComplete,
-            progressCallback: onProgress,
-            errorCallback: onError,
-            provider,
-          }) as unknown as UploadTask;
-          setUploadingFile({ id, uploadTask });
-        } else {
-          uploadFile({
-            ...processedRest,
-            file: processedFile,
-            fileName: path + processedKey,
-            isResumable: false,
-            level: accessLevel,
-            completeCallback: onComplete,
-            progressCallback: onProgress,
-            errorCallback: onError,
-            provider,
-          });
-          setUploadingFile({ id });
-        }
+        resolveFile({ processFile, file, key }).then(
+          ({ file, key, ...rest }) => {
+            onUploadStart?.({ key });
+            if (isResumable) {
+              const uploadTask = uploadFile({
+                ...rest,
+                file,
+                key: path + key,
```

Comment:
Is there any concern with the value of `key` being passed here since the user may have defined a specific value for it in `processFile`?

Reply:
As in, concern with prepending the key with `path` ? That is what we had before

Reply:
Yeah, maybe out of scope of this PR but just curious